### PR TITLE
Added non-dimensional Q criterion derived qty

### PR DESCRIPTION
**Pull request by diederikb**

### CODE REVIEW

**Review by michaeljbrazell — APPROVED** [review]
looks good, let's wait for CI before merging

File: amr-wind/fvm/qcriterion_nondim.H [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+                     idx[2];
+
+                const amrex::Real S2 =
+                    2.0 * std::pow(ux, 2) + 2.0 * std::pow(vy, 2) +
+                    2.0 * std::pow(wz, 2) + std::pow(uy + vx, 2) +
+                    std::pow(vz + wy, 2) + std::pow(wx + uz, 2);
+
+                const amrex::Real W2 = std::pow(uy - vx, 2) +
+                                       std::pow(vz - wy, 2) +
+                                       std::pow(wx - uz, 2);
```
Comment (sayerhs) [inline amr-wind/fvm/qcriterion_nondim.H:84]: This code looks like a duplicate of the QCriterion class. Can these not be combined together and differentiated at the operator functions that are defined below?

File: amr-wind/fvm/fvm.H [hunk 2]
```diff
@@ -9,6 +9,7 @@
 #include "amr-wind/fvm/vorticity.H"
 #include "amr-wind/fvm/vorticity_mag.H"
 #include "amr-wind/fvm/qcriterion.H"
+#include "amr-wind/fvm/qcriterion_nondim.H"
```
Comment (michaeljbrazell) [inline amr-wind/fvm/fvm.H:12]: I think this line can be deleted now

File: amr-wind/fvm/qcriterion.H [hunk 3]
```diff
@@ -83,12 +84,17 @@ struct Qcriterion
                                        std::pow(vz - wy, 2) +
                                        std::pow(wx - uz, 2);
 
-                qcritphi(i, j, k) = 0.5 * (W2 - S2);
+				if(nondim){
+                    qcritphi(i, j, k) = 0.5 * (W2/S2-1);
```
Comment (michaeljbrazell) [inline amr-wind/fvm/qcriterion.H:88]:
```suggestion
                    qcritphi(i, j, k) = 0.5 * (W2/S2-1.0);
```

File: amr-wind/physics/CMakeLists.txt [hunk 4]
```diff
@@ -7,6 +7,7 @@ target_sources(${amr_wind_lib_name}
   RayleighTaylor.cpp
   RayleighTaylorFieldInit.cpp
   TaylorGreenVortex.cpp
+  VortexRing.cpp
```
Comment (michaeljbrazell) [inline amr-wind/physics/CMakeLists.txt:10]:
```suggestion
```

**Pull request merged**